### PR TITLE
[SPARK-8615][Documentation]Fixed Sample deprecated code

### PR DESCRIPTION
**Pull request by tijoparacka**
Modified the deprecated jdbc api in the documentation.

### CODE REVIEW

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -1773,9 +1773,9 @@ the Data Sources API.  The following options are supported:
 <div data-lang="scala"  markdown="1">
 
 {% highlight scala %}
-val jdbcDF = sqlContext.load("jdbc", Map(
-  "url" -> "jdbc:postgresql:dbserver",
-  "dbtable" -> "schema.tablename"))
+val jdbcDF = sqlContext.read.format("jdbc").options( 
+  Map("url" -> "jdbc:postgresql:dbserver",
+  "dbtable" -> "schema.tablename")).load
```
Comment (srowen) [inline docs/sql-programming-guide.md]: Nit: `load()`

File: docs/sql-programming-guide.md [hunk 2]
```diff
@@ -1798,8 +1798,7 @@ DataFrame jdbcDF = sqlContext.load("jdbc", options)
 
 {% highlight python %}
 
-df = sqlContext.load(source="jdbc", url="jdbc:postgresql:dbserver", dbtable="schema.tablename")
-
+df = sqlContext.read.format('jdbc').options(url = 'jdbc:postgresql:dbserver', dbtable='schema.tablename').load()
```
Comment (srowen) [inline docs/sql-programming-guide.md]: I may learn something but does this syntax work? it looks like it expects a `Map`. I am just eyeballing the docs though so could be wrong

File: docs/sql-programming-guide.md [hunk 3]
```diff
@@ -1808,8 +1807,7 @@ df = sqlContext.load(source="jdbc", url="jdbc:postgresql:dbserver", dbtable="sch
 
 {% highlight r %}
 
-df <- loadDF(sqlContext, source="jdbc", url="jdbc:postgresql:dbserver", dbtable="schema.tablename")
-
+df <- read.df(sqlContext, source = "jdbc",url="jdbc:postgresql:dbserver", dbtable="schema.tablename")
```
Comment (srowen) [inline docs/sql-programming-guide.md]: Where's the ref that this was deprecated? `loadDF` doesn't appear to be

File: docs/sql-programming-guide.md (hunk 2 above)
Comment (tijoparacka) [inline docs/sql-programming-guide.md]: I tried locally its working using pyspark 
df = sqlContext.read.format('jdbc').options(url = 'jdbc:mysql://ip:port/db', driver = 'com.mysql.jdbc.Driver' ,dbtable='tablename', user ='root',password='****').load()

File: docs/sql-programming-guide.md (hunk 1 above)
Comment (tijoparacka) [inline docs/sql-programming-guide.md]: Should i update like this ?
val jdbcDF = sqlContext.read.format("jdbc").options( Map("url" -> "jdbc:postgresql:dbserver","dbtable" -> "schema.tablename")).load()

Comment (srowen) [inline docs/sql-programming-guide.md]: I'm suggesting that `load()` needs parens and you don't have them in this example. That's all

File: docs/sql-programming-guide.md (hunk 2 above)
Comment (srowen) [inline docs/sql-programming-guide.md]: Oh hah of course it's Python, nevermind. I was thinking this was another Scala example after reading the ones above.

File: docs/sql-programming-guide.md (hunk 3 above)
Comment (tijoparacka) [inline docs/sql-programming-guide.md]: you are correct I could not find any reference that loadDF is deprecated. I will revert this change back.

**Comment by andrewor14** [comment]
@yhuai

**Comment by srowen** [comment]
OK to test

**Comment by srowen** [comment]
LGTM

**Comment by liancheng** [comment]
LGTM, merging to master. Thanks for fixing this!

File: docs/sql-programming-guide.md [hunk 4]
```diff
@@ -1798,7 +1798,7 @@ DataFrame jdbcDF = sqlContext.load("jdbc", options)
 
 {% highlight python %}
 
-df = sqlContext.load(source="jdbc", url="jdbc:postgresql:dbserver", dbtable="schema.tablename")
+df = sqlContext.read.format('jdbc').options(url = 'jdbc:postgresql:dbserver', dbtable='schema.tablename').load()
```
Comment (rxin) [inline docs/sql-programming-guide.md]: python style usually don't add space before/after the = in named parameters. can you submit a follow up patch to fix that?

thanks.

Comment (tijoparacka) [inline docs/sql-programming-guide.md]: @rxin  I have submitted a followup patch with jira id -SPARK-8886.
thanks

**Pull request closed without merging**